hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketHandler.java

```diff
@@ -183,8 +183,17 @@ public static BucketHandler getBucketHandler(
                 OzoneStorageContainerManager reconSCM,
                 OmBucketInfo bucketInfo) throws IOException {
 
-    return new FSOBucketHandler(reconNamespaceSummaryManager,
+    if (bucketInfo.getBucketLayout()
```

Comment:
This version of getBucketHandler will throw an NPE if bucketInfo is null.  The other version just returns [null](https://github.com/apache/ozone/blob/39c0c25c07e359a1e5f0968a2143db1dc9da3c19/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketHandler.java#L210).  They should do the same thing, probably both throw.

Reply:
We need the other version to return null, so that we know that the BucketHandler doesn't exist and later return `EntityType.UNKNOWN`. See [here](https://github.com/xBis7/ozone/blob/HDDS-7121/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/EntityHandler.java#L149-L158).

Reply:
It's usually good practice to keep the methods consistent.  If the null check is required, it would be better to put it in the one that is called by this one, so they both have it.